src/bot/commands/crypto.js

```diff
@@ -1,21 +1,49 @@
 import { command } from '../lib/utils/command.js'
+import { currencies } from '../lib/utils/currencies.js'
 import price from 'crypto-price'
 
-async function resolver({ command: { args } }) {
-  const [cryptoCurrency = 'BTC', currency = 'USD', num = 1] = args
+// Declare it as a function and export for test purposes
+export const handleCrypto = async (cryptoCurrency, currency, quantity) => {
+  if (typeof cryptoCurrency !== 'string') {
+    return `${cryptoCurrency} is not a valid currency`
+  }
+  if (typeof currency !== 'string') {
+    return `${currency} is not a valid currency`
+  }
+
+  cryptoCurrency = cryptoCurrency.toUpperCase()
+  currency = currency.toUpperCase()
+
+  const tQuantity = quantity // For logging purpose
+  if (typeof quantity === 'string') {
+    quantity = parseFloat(quantity.replace(/,/, '.'))
+    if (isNaN(quantity)) {
+      return `${tQuantity} is not a valid number`
+    }
+  }
 
   try {
     const { base, price: cost, target } = await price.getCryptoPrice(currency, cryptoCurrency)
 
-    return `${num} ${base} is worth ${Intl.NumberFormat(currency === 'EUR' ? 'fr-FR' : 'en-US', {
+    if (currencies.indexOf(target) === -1) {
+      return `${quantity} ${base} is worth ${(cost * quantity).toFixed(2)} ${target}`
+    }
+
+    return `${quantity} ${base} is worth ${Intl.NumberFormat(currency === 'EUR' ? 'fr-FR' : 'en-US', {
```

Comment:
Could we extract the “formatting depending on the existence of a currency” in an external function?

Reply:
Formatting is extracted into `lib/utils/currencies.js`

Jest is now part of the project. 

And the full formatting and output is tested.

We need to improve repository by adding jest testing in actions like described in #7 and started in #27

Reply:
Awesome!